www/include/configuration/configObject/host_categories/DB-Func.php

```diff
@@ -85,95 +86,174 @@ function shouldNotBeEqTo0($value)
+            $bindParams = [];
             foreach ($row as $key2 => $value2) {
-                (isset($key2) && $key2 == "hc_name") ? ($hc_name = $value2 = $value2 . "_" . $i) : null;
-                $val
-                    ? $val .= ($value2 != null ? (", '" . $value2 . "'") : ", NULL")
-                    : $val .= ($value2 != null ? ("'" . $value2 . "'") : "NULL");
+                switch ($key2) {
+                    case 'hc_name':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $hc_name = $value2 = $value2 . "_" . $i;
+                        $bindParams[':hc_name'] = [\PDO::PARAM_STR => $value2];
+                        break;
+                    case 'hc_alias':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $bindParams[':hc_alias'] = [\PDO::PARAM_STR => $value2];
+                        break;
+                    case 'level':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        if ($value2) {
+                            $bindParams[':level'] = [\PDO::PARAM_INT => $value2];
+                            $level = true;
+                        } else {
+                            $bindParams[':level'] = [\PDO::PARAM_NULL => null];
+                        }
+                        break;
+                    case 'icon_id':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        $value2
+                            ? $bindParams[':icon_id'] = [\PDO::PARAM_INT => $value2]
+                            : $bindParams[':icon_id'] = [\PDO::PARAM_NULL => null];
+                        break;
+                    case 'hc_comment':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $value2
+                            ? $bindParams[':hc_comment'] = [\PDO::PARAM_STR => $value2]
+                            : $bindParams[':hc_comment'] = [\PDO::PARAM_NULL => null];
+                        break;
+                    case 'hc_activate':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_REGEXP, [
+                            "options" => [
+                                "regexp" => "/^0|1$/"
+                            ]
+                        ]);
+                        $value2
+                            ? $bindParams[':hc_activate'] = [\PDO::PARAM_STR => $value2]
+                            : $bindParams[':hc_activate'] = [\PDO::PARAM_STR => "0"];
+                        break;
+                }
+
                 if ($key2 != "hc_id") {
                     $fields[$key2] = $value2;
                 }
-                if ($key2 == "level" && $value2 != "") {
-                    $level = true;
-                }
+                $val
+                    ? $val .= ($value2 != null ? (", '" . $value2 . "'") : ", NULL")
```

Comment:
isn't $val always null ? you don't seem to have the usage of $value2 concatenation anymore.
Maybe you could replace line 229 : $val by $value2 instead